rbi/stdlib/matrix.rbi

```diff
@@ -0,0 +1,577 @@
+  #
+  # ```
+  # Vector.independent?(Vector[1,0], Vector[0,1])
+  #   => true
+  #
+  # Vector.independent?(Vector[1,2], Vector[2,4])
+  #   => false
+  # ```
+  def self.independent?(*vs); end
+
+  # Return a zero vector.
+  #
+  # ```
+  # Vector.zero(3) => Vector[0, 0, 0]
+  # ```
+  def self.zero(size); end
+
+  # Multiplies the vector by `x`, where `x` is a number or a matrix.
+  def *(x); end
+
+  # [`Vector`](https://docs.ruby-lang.org/en/2.6.0/Vector.html) addition.
+  def +(x); end
+
+  def +@(x); end
+
+  # [`Vector`](https://docs.ruby-lang.org/en/2.6.0/Vector.html) subtraction.
+  def -(x); end
+
+  def -@(x); end
+
+  # [`Vector`](https://docs.ruby-lang.org/en/2.6.0/Vector.html) division.
+  def /(x); end
+
+  # Returns `true` iff the two vectors have the same elements in the same order.
+  def ==(other); end
+
+  # Returns an angle with another vector. Result is within the [0..Math::PI].
+  #
+  # ```ruby
+  # Vector[1,0].angle_with(Vector[0,1])
+  # # => Math::PI / 2
+  # ```
+  def angle_with(v); end
+
+  def clone(); end
+
+  # The coerce method provides support for Ruby type coercion. This coercion
+  # mechanism is used by Ruby to handle mixed-type numeric operations: it is
+  # intended to find a compatible common type between the two operands of the
+  # operator. See also
+  # [`Numeric#coerce`](https://docs.ruby-lang.org/en/2.6.0/Numeric.html#method-i-coerce).
+  def coerce(other); end
+
+  # Like
+  # [`Array#collect`](https://docs.ruby-lang.org/en/2.6.0/Array.html#method-i-collect).
+  #
+  # Also aliased as:
+  # [`map`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-map)
+  def collect(); end
+
+  # Collects (as in
+  # [`Enumerable#collect`](https://docs.ruby-lang.org/en/2.6.0/Enumerable.html#method-i-collect))
+  # over the elements of this vector and `v` in conjunction.
+  def collect2(v); end
+
+  # Alias for:
+  # [`[]`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-5B-5D)
+  def component(i); end
+
+  def convector(); end
+
+  # Alias for:
+  # [`cross_product`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-cross_product)
+  def cross(*vs); end
+
+  # Returns the cross product of this vector with the others.
+  #
+  # ```ruby
+  # Vector[1, 0, 0].cross_product Vector[0, 1, 0]   => Vector[0, 0, 1]
+  # ```
+  #
+  # It is generalized to other dimensions to return a vector perpendicular to
+  # the arguments.
+  #
+  # ```ruby
+  # Vector[1, 2].cross_product # => Vector[-2, 1]
+  # Vector[1, 0, 0, 0].cross_product(
+  #    Vector[0, 1, 0, 0],
+  #    Vector[0, 0, 1, 0]
+  # )  #=> Vector[0, 0, 0, 1]
+  # ```
+  #
+  #
+  # Also aliased as:
+  # [`cross`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-cross)
+  def cross_product(*vs); end
+
+  # Alias for:
+  # [`inner_product`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-inner_product)
+  def dot(v); end
+
+  # Iterate over the elements of this vector
+  def each(&block); end
```

Comment:
https://docs.ruby-lang.org/en/2.6.0/Matrix.html#method-i-each

`each` is missing a positional param. the autogenerated signatures we have for matrix at stripe flag this as an incompatible redefinition:


```
sorbet/shims/autogenerated/gems/matrix.rbi:247: Method Matrix#each redefined without matching argument count. Expected: 0, got: 1 http://go/e/4010
     247 |  def each(which=T.unsafe(nil), &blk); end
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/0983722275df15f4e2f102c872358508a4e6c497/external/com_stripe_ruby_typer/rbi/stdlib/matrix.rbi#L345: Previous definition
     345 |  def each(&block); end
            ^^^^^^^^^^^^^^^^
Errors: 1
```